superset/assets/javascripts/dashboard/v2/components/BuilderComponentPane.jsx

```diff
@@ -1,32 +1,67 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import cx from 'classnames';
 
-import NewChart from './gridComponents/new/NewChart';
 import NewColumn from './gridComponents/new/NewColumn';
 import NewDivider from './gridComponents/new/NewDivider';
 import NewHeader from './gridComponents/new/NewHeader';
 import NewRow from './gridComponents/new/NewRow';
 import NewTabs from './gridComponents/new/NewTabs';
+import SliceAdderContainer from '../../../dashboard/components/SliceAdderContainer';
 
 const propTypes = {
   editMode: PropTypes.bool,
 };
 
 class BuilderComponentPane extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showSlices: false,
+    };
+
+    this.toggleSlices = this.toggleSlices.bind(this);
+    this.openSlicesPane = this.toggleSlices.bind(this, true);
+    this.closeSlicesPane = this.toggleSlices.bind(this, false);
+  }
+
+  toggleSlices(show) {
```

Comment:
nit you could just call this `showSlices` and call `this.setState({ showSlices });`